### DISCUSSION TITLE
Adaptive update/ casync together with slot skipping

### DISCUSSION THREAD

**Discussion by tcikel** [rauc/rauc#1066] [Q&A]
Hello, 
Is it possible to use adaptive or delta updates with slot skipping? We are updating our apps quite often and rootfs almost never, so it feels like waste of time to calculate hashes for rootfs when we know it doesn't change, from my limited testing I couldn't get it working. Is this approach even good? Or am I missing something? Or is it at least possible to skip calculating hash of inactive rootfs ? I saw that there is a implementation for something like that  but not merged yet https://github.com/rauc/rauc/pull/513.

**Comment by ejoerns** [comment]
Slot skipping is independent of other mechanisms and can be combined, yes.

The adaptive update method 'block-hash-index' does not need to re-calculate hashes on the target if the images haven't changed. The information is cached in a file on the device.

For casync the hashes are calculated on the target by default. Potentially this could be replaced by cached information, too (depending on the casync variant used).

**Reply by ejoerns** [reply]
FTR: I have updated the slot skipping documentation to mention streaming in #1068
